modules/database/src/ioc/db/dbConvertJSON.c

```diff
@@ -66,39 +64,23 @@ static int dbcj_double(void *ctx, double num) {
     return 1;
 }
 
-static int dblsj_double(void *ctx, double num) {
-    return 0;    /* Illegal */
-}
-
 static int dbcj_string(void *ctx, const unsigned char *val, size_t len) {
     parseContext *parser = (parseContext *) ctx;
     char *pdest = parser->pdest;
 
-    /* Not attempting to handle char-array fields here, they need more
-     * metadata about the field than we have available at the moment.
-     */
```

Comment:
We do have all the metadata needed: parser->elems

Reply:
Array elements will be concatenated (optionally mixed with numeric single bytes) into one long string. This allows to split a long constant string  into multiple lines in the template file.

Reply:
The RELEASE_NOTES message says that concatenation works on CHAR arrays, not that it would work on STRING (where due to the limited STRING size line breaks are not so pressing). I could either modify the text so that is it more explicit about what the change does NOT do or I can implement concatenation on STRING scalars, but not STRING arrays (which I find even more confusing; what about STRING[1] arrays?)

I understand your hesitation concerning raw bytes in long strings. Implementing numbers as Unicode code points would be possible as well, but with an ambiguity for code points 0x80-0xff. On the other hand, we do not have Unicode support in EPICS anywhere but in JSON strings (where is is enforced, which I don't like because historically, we still have many Latin-1 encoded strings.)

Of course I can disable accepting numbers in long strings (but not in CHAR waveforms, which may really mean 8 bit integer numbers) but I do not really see why, as it comes for free. Where does it really hurt?

Concerning floating point numbers: CHAR (or any other integer type) waveforms already now accept floating point values.